src/D3MHub.sol

```diff
@@ -275,38 +273,43 @@ contract D3MHub {
         if (art < ink) { // If there was permissionless DAI paid or fees added as collateral
             address _vow = vow;
             uint256 fixArt = ink - art; // Amount of fees + permissionless DAI paid we will now transform to debt
-            delete art;
-            Art += fixArt;
-            require(Art <= MAXINT256, "D3MHub/overflow");
+            art += fixArt;
+            require(art <= MAXINT256, "D3MHub/overflow");
             vat.suck(_vow, _vow, fixArt * RAY); // This needs to be done to make sure we can deduct sin[vow] and vice in the next call
             vat.grab(ilk, address(_pool), address(_pool), _vow, 0, int256(fixArt)); // Generating the debt
         }
 
         // Determine if it needs to unwind or wind
 
-        uint256 Line = vat.Line();
-        uint256 debt = vat.debt();
+        uint256 Line;
+        uint256 debt;
         uint256 targetAssets;
         uint256 toUnwind;
 
-        if (ilks[ilk].tic != 0 || !ilks[ilk].plan.active()) { // If D3M is caged (but not culled) or plan is not active
+        // Determine if it needs to fully unwind due D3M ilk being caged (but not culled) or plan is not active
+        if (ilks[ilk].tic != 0 || !ilks[ilk].plan.active()) {
             toUnwind = type(uint256).max; // We make sure to enter the unwind path
         } else {
-            if (Art > lineWad) {
+            Line = vat.Line();
+            debt = vat.debt();
+            targetAssets = ilks[ilk].plan.getTargetAssets(currentAssets);
+
+            // Determine if it needs to unwind due ilk debt ceiling
+            if (art > lineWad) {
                 unchecked {
-                    toUnwind = Art - lineWad; // checks if we need to unwind due ilk debt ceiling
+                    toUnwind = art - lineWad;
                 }
             }
+            // Determine if it needs to unwind due global debt ceiling
```

Comment:
Minor English - should be "due to" in all these comments

Reply:
Fixed!